pandas/core/tools/datetimes.py

```diff
@@ -446,8 +447,15 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
             except (ValueError, TypeError):
                 raise e
 
+    def check_numerical_arg():
+        return ((is_scalar(arg) and (is_integer(arg) or is_float(arg))) or
```

Comment:
It can be a scalar argument too (example: `pd.to_datetime(2, unit='ns')` ; arg = 2 here), i have to do a np.asarray